test/OpenTelemetry.Tests/Trace/BatchExportActivityProcessorTest.cs

```diff
@@ -0,0 +1,142 @@
+﻿// <copyright file="BatchExportActivityProcessorTest.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using OpenTelemetry.Tests.Shared;
+using OpenTelemetry.Trace;
+using Xunit;
+
+namespace OpenTelemetry.Tests.Trace
+{
+    public class BatchExportActivityProcessorTest
+    {
+        [Fact]
+        public void CheckNullExporter()
+        {
+            Assert.Throws<ArgumentNullException>(() => new BatchExportActivityProcessor(null));
+        }
+
+        [Fact]
+        public void CheckConstructorWithInvalidValues()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() => new BatchExportActivityProcessor(new TestActivityExporter(), maxQueueSize: 0));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new BatchExportActivityProcessor(new TestActivityExporter(), maxExportBatchSize: 0));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new BatchExportActivityProcessor(new TestActivityExporter(), maxExportBatchSize: 2049));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new BatchExportActivityProcessor(new TestActivityExporter(), scheduledDelayMilliseconds: 0));
+            Assert.Throws<ArgumentOutOfRangeException>(() => new BatchExportActivityProcessor(new TestActivityExporter(), exporterTimeoutMilliseconds: -1));
+        }
+
+        [Fact]
+        public void CheckIfBatchIsExportingOnQueueLimit()
+        {
+            using var exporter = new TestActivityExporter();
+            using var processor = new BatchExportActivityProcessor(exporter, maxQueueSize: 1, maxExportBatchSize: 1);
```

Comment:
Since we're testing the queue limit trigger instead of timer trigger, consider setting `scheduledDelayMilliseconds` to a very big value.

Reply:
just updated! thanks for reviewing!